File: ts-sdk/src/sdk/models/operations/getusers.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";


export enum GetUsersUserTypeEnum {
    All = "all",
    Managed = "managed",
    External = "external"
}


export class GetUsersQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=external_app_user_id" })
  externalAppUserId?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=filter_term" })
  filterTerm?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=marker" })
  marker?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=offset" })
  offset?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=usemarker" })
  usemarker?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=user_type" })
  userType?: GetUsersUserTypeEnum;
}


export class GetUsersRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  queryParams: GetUsersQueryParams;
}


export class GetUsersResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  users?: shared.Users;
}

File: go-sdk/pkg/models/operations/getusers.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetUsersUserTypeEnum string

const (
	GetUsersUserTypeEnumAll      GetUsersUserTypeEnum = "all"
	GetUsersUserTypeEnumManaged  GetUsersUserTypeEnum = "managed"
	GetUsersUserTypeEnumExternal GetUsersUserTypeEnum = "external"
)

type GetUsersQueryParams struct {
	ExternalAppUserID *string               `queryParam:"style=form,explode=true,name=external_app_user_id"`
	Fields            []string              `queryParam:"style=form,explode=false,name=fields"`
	FilterTerm        *string               `queryParam:"style=form,explode=true,name=filter_term"`
	Limit             *int64                `queryParam:"style=form,explode=true,name=limit"`
	Marker            *string               `queryParam:"style=form,explode=true,name=marker"`
	Offset            *int64                `queryParam:"style=form,explode=true,name=offset"`
	Usemarker         *bool                 `queryParam:"style=form,explode=true,name=usemarker"`
	UserType          *GetUsersUserTypeEnum `queryParam:"style=form,explode=true,name=user_type"`
}

type GetUsersRequest struct {
	Retries     *utils.RetryConfig
	QueryParams GetUsersQueryParams
}

type GetUsersResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
	Users       *shared.Users
}
